### PR TITLE
feature: Added snmp_check() to decide if a device is up/down for snmp #5946

### PR DESCRIPTION
**Pull request by laf**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

Pretty sure this fixes: #5946 

It falls back to the old way if this doesn't work (or should). Removed the hack for hikvision as it shouldn't have been done like that.

The idea behind this is that we only care if we have a response code of 0 from the call, which any lookup that gets a form of snmp response has. I checked with snmpsimd and added a file with just sysDescr response, this is detected as being ok and returns true (we don't care we have a sysObjectId we just want to know snmp is up).

### CODE REVIEW

File: includes/snmp.inc.php [hunk 1]
```diff
@@ -255,6 +255,25 @@ function snmp_get($device, $oid, $options = null, $mib = null, $mibdir = null)
     }
 }//end snmp_get()
 
+/**
+ * @param $device
+ * @return bool
+ */
+function snmp_check($device)
+{
+    $time_start = microtime(true);
```
Comment (murrant) [inline includes/snmp.inc.php]: $time_start is unused ;)

File: includes/functions.php [hunk 2]
```diff
@@ -557,15 +557,16 @@ function isSNMPable($device)
...
-        // Support for Hikvision
-        $pos = snmp_get($device, "SNMPv2-SMI::enterprises.39165.1.1.0", "-Oqv", "SNMPv2-MIB");
-    }
-    if ($pos === '' || $pos === false) {
-        return false;
-    } else {
+    $pos = snmp_check($device);
+    if ($pos === true) {
         return true;
+    } else {
```
Comment (murrant) [inline includes/functions.php]: Why try two times?

Comment (laf) [inline includes/functions.php]: For now it's catch all. Whilst I'm sure this change will work I can't be 100% sure, all I've been able to test against is net-snmpd and snmpsimd. Up / down is the bit I know works but actually checking we have sysObjectId is where I've had limited testing.

We can remove it but I just didn't want to.

File: includes/snmp.inc.php (hunk 1 above)
Comment (laf) [inline includes/snmp.inc.php]: Didn't mean to remove the line that used it, added that back in.

**Pull request merged**